### ISSUE TITLE
Consider capturing git the line break configuration in sourcelink json

### ISSUE THREAD

**Issue by tmat** [dotnet/sourcelink#613]
This change in .NET SDK added assembly-level attribute with git repo clone URL: https://github.com/dotnet/sdk/pull/11559

This can be used to optimize download of all source file required to build the dll. If however, the git configuration (specifically `core.eol` and `core.autocrlf` settings) differ between the server that originally built the DLL and the local repo the re-build will produce different binaries.

It would thus be useful to record the git settings in Source Link record so that the clone can be configured the same as the build server that produced the binary.

**Comment by tmat** [comment]
@clairernovotny

**Comment by bording** [comment]
I think the situation is more involved than those settings you've mentioned. If the repo has a `.gitattributes` file defined, the values in it override any local configuration settings.

**Comment by tmat** [comment]
Yes, but `.gitattributes` would be cloned with the repo clone. We only need to capture configuration that's not stored on the server.

**Comment by bording** [comment]
> We only need to capture configuration that's not stored on the source server.

I guess I'm missing when you'd think those values would matter. Are you thinking of cases where the repo doesn't have a `.gitattributes` file?

**Comment by tmat** [comment]
Yes, also in case when `.gitattributes` doesn't specify line breaks for all files that are input to the compiler.
